x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -162,7 +162,10 @@ public enum DataType {
 
     static {
         Map<String, DataType> map = TYPES.stream().filter(e -> e.esType() != null).collect(toMap(DataType::esType, t -> t));
-        map.put("date_nanos", DATETIME);
```

Comment:
To explain a bit about what's going on with date nanos here - this mapping was not being used by ES|QL, as we use the lookup from `EsqlDataTypes` (which is what this PR is merging into `DataType`).  I presume this is a left over mapping from SQL.  We also don't want to treat date nanos as datetimes, since that will cause weird type compatibility issues.  So, removing this legacy mapping is actually a step towards supporting nanosecond dates.

Reply:
:+1: